nixos/modules/config/xdg/portal.nix

```diff
@@ -48,23 +48,29 @@ with lib;
       packages = [ pkgs.xdg-desktop-portal ] ++ cfg.extraPortals;
       joinedPortals = pkgs.symlinkJoin {
         name = "xdg-portals";
-        paths = cfg.extraPortals;
+        paths = packages;
       };
 
-    in mkIf cfg.enable {
+    in
+    mkIf cfg.enable {
 
       assertions = [
-        { assertion = (cfg.gtkUsePortal -> cfg.extraPortals != []);
-          message = "Setting xdg.portal.gtkUsePortal to true requires a portal implementation in xdg.portal.extraPortals such as xdg-desktop-portal-gtk or xdg-desktop-portal-kde.";
+        {
+          assertion = cfg.extraPortals != [ ];
+          message = "Setting xdg.portal.enable to true requires a portal implementation in xdg.portal.extraPortals such as xdg-desktop-portal-gtk or xdg-desktop-portal-kde.";
         }
       ];
 
-      services.dbus.packages  = packages;
+      services.dbus.packages = packages;
       systemd.packages = packages;
 
       environment.sessionVariables = {
         GTK_USE_PORTAL = mkIf cfg.gtkUsePortal "1";
         XDG_DESKTOP_PORTAL_DIR = "${joinedPortals}/share/xdg-desktop-portal/portals";
+        # fixes screen sharing on plasmawayland on non-chromium apps by linking
+        # share/applications/*.desktop files
+        # see https://github.com/NixOS/nixpkgs/issues/145174
+        XDG_DATA_DIRS = [ "${joinedPortals}/share" ];
```

Comment:
This is literally the same as `environment.systemPackages` except with extra value in `XDG_DATA_DIRS`.

Reply:
Isn't it clearer this way? Or is setting systemPackages preferred?

Reply:
I meant same in the way that it pulls in other files than just `share/applications`

Reply:
ok now they only link to the needed directories
This should answer @jtojnar 's objection right?

Reply:
Actually, thinking about this a bit more, more items in `XDG_DATA_DIRS` is problematic because it further increases various lookup times (see e.g. https://github.com/NixOS/nixpkgs/issues/156830). So I would go with

```nix
environment.systemPackages = [ joinedPortals ];
environment.pathsToLink = [ "share/applications" ];
```

instead of this line, after all. Everything else can remain the same.

Reply:
does it really need the `environment.pathsToLink` since that is in the `joinedPortals` env already?

Reply:
I would expect the system-wide `buildEnv` to consider `joinedPortals` a package like any other so I would think so (still with the DEs’ `/share` caveat).

Reply:
ok I've committed it to systemPackages